src/Filesystem/F.php

```diff
@@ -860,6 +862,49 @@ public static function write(string $file, mixed $content, bool $append = false)
 			throw new Exception('The file "' . $file . '" is not writable');
 		}
 
-		return file_put_contents($file, $content, $mode) !== false;
+		// the name of the temporary file it will write
+		// to before it starts overwriting the original file
+		$temp = $file . '~';
+
+		// @codeCoverageIgnoreStart
+		$result = Helpers::handleErrors(
+			fn (): bool => file_put_contents($temp, $content, $mode) !== false,
+			function (&$override, int $errno, string $errstr) use ($temp): bool {
+				// ensure the temp file gets deleted even
+				// if an exception gets thrown
+				F::unlink($temp);
+
+				// consider an exceeded disk quota as a hard error
+				// to make the issue visible to the user
+				if (
+					Str::contains($errstr, 'error 122', true) ||
+					Str::contains($errstr, 'errno=122', true) ||
+					Str::contains($errstr, 'Quota exceeded', true) ||
+					Str::contains($errstr, 'Disk quota exceeded', true)
+				) {
+					throw new Exception($errstr);
+				}
+
+				// handle every other warning normally
+				return false;
+			}
+		);
+		// @codeCoverageIgnoreEnd
+
+		try {
+			// if everything is ok, rename temporary file with original
+			if ($result === true) {
+				rename($temp, $file);
+
+				return true;
+			}
+		} catch (Throwable) {
```

Comment:
When will `rename()` throw an exception? I believe the exception again comes from Whoops only.

Reply:
I'm getting error on `testWriteHandleWarning` unit test without try catch block. Do you?

Reply:
````
rename(\kirby\tests\Filesystem/tmp/test~,\kirby\tests\Filesystem/tmp/test): Access Denied (code: 5)
 \kirby\src\Filesystem\F.php:896
 \kirby\tests\Filesystem\FTest.php:847
 ````

Reply:
That's likely a warning and not an exception though. For PHP warnings and errors we need the error handler. We cannot catch these with `try catch` if Whoops is disabled.

Reply:
Yes, this is warning.